core/src/main/java/org/infinispan/protostream/impl/TagReaderImpl.java

```diff
@@ -1102,6 +1099,12 @@ private void skipNBytes(long n) throws IOException {
             }
          }
       }
+
+      @Override
+      protected Decoder decoderFromLength(int length) throws IOException {
+         byte[] bytes = readRawByteArray(length);
+         return new ByteArrayDecoder(bytes, 0, length);
```

Comment:
I believe this is what I tried before and it doesn't work as the in bytes are not consumed, but I can try it again to verify.

Reply:
Yeah, adding this change I get a failure in the test suite. This is because the way the parser works is the sub decoder isn't read until after all the other bytes are read. So if there is another object after the decoder then it would fail as it is reading the wrong spot. Luckily our code will not be using this in the long term, so we won't have this extra allocation.